### PR TITLE
doc: Added documentation for AD719X project

### PR DESCRIPTION
**Pull request by D-Disha**
Added sphinx documentation.

### CODE REVIEW

**Comment by SaikiranGudla** [comment]
Looks good to me.
Suggest to squash commits by not adding extra commits for review changes. 
A changelog in the PR discussion thread similar to no-os would be better.

**Comment by D-Disha** [comment]
Added documentation support for ad7124 console application. 
The wiki link for ad7124 console : https://wiki.analog.com/resources/tools-software/product-support-software/ad7124_mbed_support

File: doc/sphinx/source/projects/ad7124_console/ad7124_console.rst [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+Temperature sensing application using AD7124 can be found `here <https://www.analog.com/media/en/reference-design-documentation/reference-designs/CN0383.pdf>`_
+
+====================
+AD7124 Mbed Firmware
+====================
+This section briefs on the usage of MBED firmware. This also explains the steps to compile and build the application using
+mbed and make based build.
+
+   .. image:: /source/projects/ad7124_console/ad7124_software_sequence.jpg
```
Comment (SaikiranGudla) [inline doc/sphinx/source/projects/ad7124_console/ad7124_console.rst:111]: Shall the image be attached after the line 118 following the description written which says "AD7124 Firmware Example is **shown below**" (line114) ?

File: doc/sphinx/source/projects/ad7124_console/ad7124_console.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+.. Project Build Section:
+    
+.. include:: /source/build/project_build.rst
+
+===========
+Quick Start
+===========
+
+If you have some familiarity with the Mbed platform, the following is a basic list of steps required to start running the code,
```
Comment (SaikiranGudla) [inline doc/sphinx/source/projects/ad7124_console/ad7124_console.rst:128]: I think Quick Start needs to be modified a bit as building and flashing the binary is already covered in the Build section.
Lines 128 to 137 are redundant and there isn't a Downloads section as mentioned in line 133.

**Review by SaikiranGudla — CHANGES_REQUESTED** [review]

**Comment by D-Disha** [comment]
Changelog: Updated Quick start and moved the software layers image

**Review by SaikiranGudla — APPROVED** [review]

**Comment by D-Disha** [comment]
Hi @CMinajigi @ribdp, are there any more comments or does this PR look good for merge?

File: doc/sphinx/source/projects/ad7124_console/ad7124_console.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+the hardware dependency as any Mbed enabled board can be used with same firmware with little modifications (precisely changing a pin mapping).
+
+=================
+Interface Diagram
+=================
+
+   .. image:: /source/projects/ad7124_console/ad7124_interface_diagram.jpg
+      :width: 500
+
+The AD7124-EVAL board is connected to SDP-K1 through default 120-pin SDP Connector. The AD7124-EVAL board has external connector to connect
```
Comment (ribdp) [inline doc/sphinx/source/projects/ad7124_console/ad7124_console.rst:51]: Should we leave out 'default' here?

File: doc/sphinx/source/projects/ad717x_console/ad717x_console.rst [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+**Supported Carrier Boards:**
+
+* `SDP-K1 With Mbed <https://os.mbed.com/platforms/SDP_K1/>`_
+
+============
+Introduction
+============
+
+The AD717x/ AD411x family offer a complete integrated Sigma-Delta ADC solution which can be used in high precision, low noise single channel applications (Life Science measurements) or higher speed multiplexed applications (Factory Automation PLC Input modules).
```
Comment (SaikiranGudla) [inline doc/sphinx/source/projects/ad717x_console/ad717x_console.rst:32]: nit: remove extra space after "/"

File: doc/sphinx/source/projects/ad717x_console/ad717x_console.rst [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+the hardware dependency as any Mbed enabled board can be used with same firmware with little modifications (precisely changing a pin mapping).
+
+=================
+Interface Diagram
+=================
+
+   .. image:: /source/projects/ad717x_console/ad717x_interface_diagram.jpg
+      :width: 500
+
+The EVAL-AD411x/ EVAL-AD717x board is connected to SDP-K1 through on-board default 120-pin SDP Connector. The AD717x/AD411x EVAL boards can be powered-up through a SDP-K1 USB supply or from external DC supply,
```
Comment (SaikiranGudla) [inline doc/sphinx/source/projects/ad717x_console/ad717x_console.rst:55]: nit: remove extra space after "/"

**Pull request merged**